clients/adapter-kafka/src/main/java/org/eclipse/hono/adapter/client/command/kafka/KafkaBasedCommandContext.java

```diff
@@ -57,21 +61,34 @@ public KafkaBasedCommand getCommand() {
 
     @Override
     public void accept() {
+        if (!completed.compareAndSet(false, true)) {
```

Comment:
out of curiosity: why are you guarding this with a boolean?

Reply:
The guard is mainly for the other methods (reject/modify/release). It's a preparation for when in the KafkaBasedCommandContext class a command response is to be sent, making sure it doesn't get sent twice.
(In the AMQP context, invoking these methods twice has no effect, so there might be a case where this may happen.)

Reply:
I see. Aren't you a little worried that two *different* methods might be invoked instead of the same method being invoked twice? This would raise the question of which invocation actually reflects the proper outcome, right?

Reply:
I'm thinking more about cases where "released" might be invoked twice. I don't know about actual cases where this may happen, but I'm not 100% certain it can't happen (cases where a timeout for getting device registry info has kicked in but command processing was already cancelled because the corresponding command partition got unassigned, something like that).
So this is just a safety measure. I've also just changed the code here a bit, logging both outcomes.